datafusion/physical-expr/src/aggregate/median.rs

```diff
@@ -111,13 +112,28 @@ impl PartialEq<dyn Any> for Median {
 /// The intermediate state is represented as a List of those scalars
 struct MedianAccumulator {
     data_type: DataType,
+    batches: Vec<ArrayRef>,
```

Comment:
Nit is that this is actually arrays (not batches)  -- maybe `arrays` is a better name